lib/internal/modules/esm/hooks.js

```diff
@@ -497,13 +438,106 @@ class Hooks {
       __proto__: null,
       format,
       responseURL,
-      source,
+      source: ''+source,
     };
   }
 }
-
 ObjectSetPrototypeOf(Hooks.prototype, null);
 
+class HooksProxy {
+  /**
+   * The request & response segment of the shared memory. TextEncoder/Decoder (needed to convert
+   * requests & responses into a format supported by the comms channel) reads and writes with
+   * Uint8Array.
+   */
+  #data;
+  /**
+   * The lock/unlock segment of the shared memory. Atomics require this to be a Int32Array. This
+   * segment is used to tell the main to sleep when the worker is processing, and vice versa
+   * (for the worker to sleep whilst the main thread is processing).
+   * 0 -> main sleeps
+   * 1 -> worker sleeps
+   */
+  #lock;
+
+  #isReady = false;
+
+  /**
+   * @param {import('./loader.js).KeyedExports} customLoaders Exports from user-defined loaders
+   *  (as returned by `ModuleLoader.import()`).
+   */
+  constructor() {
+    const { InternalWorker } = require('internal/worker');
+
+    const data = new SharedArrayBuffer(204_800);
+    const lock = new SharedArrayBuffer(4);
+    this.#data = new Uint8Array(data);
+    this.#lock = new Int32Array(lock);
```

Comment:
We should use only one SAB (but also it's an optimization that we can do later, it'd be fine to add a TODO)
```suggestion
    const data = new SharedArrayBuffer(204_804);
    this.#data = new Uint8Array(data, 1);
    this.#lock = new Int32Array(data, 0, 1);
```